src/native/corehost/corehost.cpp

```diff
@@ -84,12 +84,29 @@ bool is_exe_enabled_for_execution(pal::string_t* app_dll)
 #define CURHOST_EXE
 #endif
 
-void need_newer_framework_error()
+void need_newer_framework_error(const pal::string_t& dotnet_root)
 {
-    pal::string_t url = get_download_url();
-    trace::error(_X("  _ To run this application, you need to install a newer version of .NET Core."));
```

Comment:
Can we get rid of the check for the `  _ ` prefix now?
https://github.com/dotnet/runtime/blob/febeba309d6606f2c3ca0ee5ef4ba161596e8a25/src/native/corehost/apphost/apphost.windows.cpp#L116

Reply:
Isn't it necessary for the pre 7.0 runtimes?
7.0 doesn't use it anymore, but the apphost still needs to work downlevel for some rather weird cases.

Reply:
I thought apphost itself was the only thing actually writing with that prefix? Because with downlevel hostfxr, we couldn't actually redirect the errors from hostfxr.

Reply:
What about 7.0 `apphost` running against 6.0 `hostfxr` - the `hostfxr` might still use the `_` prefix somewhere (I didn't check). In that case the error info will be redirected from `hostfxr` to `apphost` and if it contains `_` it should be parsed.

Reply:
I was pretty sure we only ever used it for `apphost`, but I didn't actually go back and check.

Either way, the UX and changes you have right now look good to me.

Reply:
I just searched 3.1 and it only uses the prefix in apphost - but I'm still nervous about doing this. Honestly I would rather keep it.